virtool/users/sessions.py

```diff
@@ -2,219 +2,220 @@
+            raise ResourceError()
 
-    hashed_token = hashlib.sha256(session_token.encode()).hexdigest()
-    if stored_session_token == hashed_token:
-        return session, session_token
-
-
-async def create_reset_code(
-    redis: Redis, session_id: str, user_id: str, remember: Optional[bool] = False
-) -> str:
-    """
-    Create a secret code that is used to verify a password reset request. Properties:
-
-    - the reset request must pass a reset code that is associated with the session
-      linked to the request
-    - the reset code is dropped from the session for any non-reset request sent after
-      the code was generated
-
-    :param redis:
-    :param session_id:
-    :param user_id:
-    :param remember:
-    :return:
-
-    """
-    reset_code = secrets.token_hex(32)
-
-    unparsed_session, session_expiry = await asyncio.gather(
-        redis.get(session_id), redis.ttl(session_id)
-    )
-
-    session = json.loads(unparsed_session)
-
-    await redis.set(
-        session_id,
-        dumps(
-            {
-                **session,
-                "reset_code": reset_code,
-                "reset_remember": remember,
-                "reset_user_id": user_id,
-            }
-        ),
-        expire=session_expiry,
-    )
-
-    return reset_code
-
-
-async def clear_reset_code(redis: Redis, session_id: str):
-    """
-    Clear the reset information attached to the session associated with the passed
-    `session_id`.
-
-    :param redis: the application redis client
-    :param session_id: the session id
-
-    """
-
-    unparsed_session, session_expiry = await asyncio.gather(
-        redis.get(session_id), redis.ttl(session_id)
-    )
-
-    if not unparsed_session:
-        return
-
-    session = json.loads(unparsed_session)
-
-    await redis.set(
-        session_id,
-        dumps(
-            {
-                key: session[key]
-                for key in session
-                if key not in {"reset_code", "reset_remember", "reset_user_id"}
-            }
-        ),
-        expire=session_expiry,
-    )
-
-
-async def replace_session(
-    db: DB,
-    redis: Redis,
-    session_id: str,
-    ip: str,
-    user_id: Optional[str] = None,
-    remember: Optional[bool] = False,
-) -> Tuple[str, dict, str]:
-    """
-    Replace the session associated with `session_id` with a new one. Return the new
-    session document.
-
-    Supplying a `user_id` indicates the session is authenticated. Setting `remember`
-    will make the session last for 30 days instead of the default 30 minutes.
-
-    :param db: the application database client
-    :param redis: the application redis pool
-    :param session_id: the id of the session to replace
-    :param ip:
-    :param user_id:
-    :param remember:
-    :return: new session document and token
-    """
-    await redis.delete(session_id)
-    return await create_session(db, redis, ip, user_id, remember=remember)
+        return reset.user_id, reset.remember
```

Comment:
Yeah I think your right. I initially avoided it since returning the reset code is redundant, but the defined interface is more than worth it. Changed in local version